airbyte-integrations/connectors/source-okta/source_okta/custom_authenticators.py

```diff
@@ -61,3 +64,55 @@ def refresh_access_token(self) -> Tuple[str, int]:
             return response_json["access_token"], response_json["expires_in"]
         except Exception as e:
             raise Exception(f"Error while refreshing access token: {e}") from e
+
+
+@dataclass
+class CustomOauth2PrivateKeyAuthenticator(DeclarativeAuthenticator):
+    """
+    Custom authenticator that uses a signed JWT with a private key to authenticate against Okta.
+    """
+
+    config: Config
+
+    @property
+    def auth_header(self) -> str:
+        return "Authorization"
+
+    @property
+    def token(self) -> str:
+        domain = self.config["domain"]
+        client_id = self.config["credentials"]["client_id"]
```

Comment:
I don't know exactly how, but maybe there's a nice way to provide graceful errors if credentials are not provided for any reason? Or will it just fails the `spec` call? /cc @artem1205 @darynaishchenko